main.py

```diff
@@ -0,0 +1,76 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from app import schemas, crud
+from app.database import SessionLocal
+
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+    skip: int | None = None,
+    limit: int | None = None,
+    db: Session = Depends(get_db)
+):
+    return crud.get_all_authors(db=db, skip=skip, limit=limit)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Author already exists"
+        )
+
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_single_author(author_id: int, db: Session = Depends(get_db)):
+    if db_author := crud.get_author(db=db, author_id=author_id):
+        return db_author
+
+    raise HTTPException(
+        status_code=404,
+        detail="Author not found"
+    )

```

Comment:
This functionality is not implemented "Filter books by author ID."

Reply:
![image](https://github.com/mate-academy/py-fastapi-library-management-api/assets/87121473/e10124ce-76fa-4024-a880-c9c2e7d80c20)